src/store.rs

```diff
@@ -325,9 +325,12 @@ impl<F: LurkField> Copy for ScalarPtr<F> {}
 
 impl<F: LurkField> Display for ScalarPtr<F> {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        let trimmed_f1 = self.0.trimmed_hex_digits();
+        let tag_str = match ExprTag::from_field(&self.0) {
```

Comment:
If you rebase, you’ll see it’s probably useless to use `from_field` as that first element is already a tag. You may get away with implementing this on the sole `SPtr` generic.